### PR TITLE
feat(egh): prompt to share on publish

### PR DESCRIPTION
**Pull request by zacjones93**
This provides a CTA when instructors publish a post. This is done by pushing a query param `published=true` using the `onPublished` callback in  `EditResourcesFormDesktop`

image

gif

### CODE REVIEW

File: apps/egghead/src/app/(content)/posts/[slug]/edit/actions.ts [hunk 1]
```diff
@@ -43,3 +43,7 @@ export const onPostSave = async (resource: ContentResource) => {
 	revalidatePath(`/${post?.fields?.slug}`)
 	redirect(`/${resource.fields?.slug}`)
 }
+
+export const onPostPublish = async (resource: ContentResource) => {
+	redirect(`/${resource.fields?.slug}?published=true`)
```
Comment (zacjones93) [inline apps/egghead/src/app/(content)/posts/[slug]/edit/actions.ts:48]: thought query param would be best way to show modal when someone publishes

File: apps/egghead/src/app/_components/share-post-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		const published = searchParams.get('published')
+		if (published === 'true') {
+			setOpen(true)
+		}
+	}, [searchParams])
+
+	const handleClose = () => {
+		setOpen(false)
+		router.push(window.location.pathname)
+	}
```
Comment (zacjones93) [inline apps/egghead/src/app/_components/share-post-modal.tsx:39]: remove query param when modal is closed

File: apps/egghead/src/app/_components/share-post-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+					<Share post={post} message={message} />
+					<DialogDescription className="flex items-center justify-between pt-4 text-sm">
+						<span>Add a message to your post</span>{' '}
+						<span>({message.length}/300)</span>
+					</DialogDescription>
+					<Textarea
+						value={message}
+						onChange={(e) => setMessage(e.target.value)}
+						placeholder="Learn how to..."
+					/>
```
Comment (zacjones93) [inline apps/egghead/src/app/_components/share-post-modal.tsx:58]: thought might be nice to write out a social post in one form and have that copy over to X and bluesky (linkedin doesn't support pre-filling post body)

File: packages/ui/resources-crud/edit-resources-form-desktop.tsx [hunk 4]
```diff
@@ -91,6 +92,10 @@ export function EditResourcesFormDesktop({
 		action: 'save' | 'publish' | 'archive' | 'unpublish' = 'save',
 	) => {
 		const updatedResource = await updateResource(values, action)
+		if (action === 'publish' && onPublish) {
+			return await onPublish(updatedResource)
+		}
+
```
Comment (zacjones93) [inline packages/ui/resources-crud/edit-resources-form-desktop.tsx:98]: needed to actually wire up the `onPublish` hook being called

**Review by nicollguarnizo — APPROVED** [review]

**Pull request merged**